packages/framework/esm-react-utils/src/useDebounce.ts

```diff
@@ -0,0 +1,43 @@
+import { useEffect, useState } from "react";
+
+/**
+ * This hook will help us out in getting the debounced value of a state variable,
+ * which then makes a request to the backend.
+ * 
+ * For example,
+ * 
+ * ```tsx
+ * import { useDebounceValue } from "@openmrs/esm-react-utils";
+ * 
+ * function MyComponent() {
+ *   const [searchTerm, setSearchTerm] = useState('');
+ *   const debouncedSearchTerm = useDebounce(searchTerm);
+ *   const swrResult = useSWR(`/api/search?q=${debouncedSearchTerm}`) 
+ * 
+ *  return (
+ *    <Search
+        labelText={t('search', 'Search')}
+        onChange={(e) => setSearchTerm(e.target.value)}
+        value={searchTerm}
+      />
+ *  )
+ * }
+ * 
+ * ```
+ * 
+ * @param value
+ * @param delay number = 300
+ * @returns debounceValue
+ */
```

Comment:
```suggestion
 * ```
 * 
 * @param value: The value that will be used to set `debounceValue`
 * @param delay number = 300: The number of milliseconds to wait before updating `debounceValue`
 * @returns debounceValue: The debounced value
 */
```